lib/activerecord-multi-tenant/multi_tenant.rb

```diff
@@ -24,13 +24,24 @@ def self.enable_with_lock_workaround; @@enable_with_lock_workaround = true; end
   def self.with_lock_workaround_enabled?; @@enable_with_lock_workaround; end
 
   # Registry that maps table names to models (used by the query rewriter)
-  def self.register_multi_tenant_model(table_name, model_klass)
-    @@multi_tenant_models ||= {}
-    @@multi_tenant_models[table_name.to_s] = model_klass
+  # @deprecated _table_name is no longer used
+  def self.register_multi_tenant_model(_table_name, model_klass)
+    @@multi_tenant_models ||= []
+    @@multi_tenant_models.push(model_klass)
```

Comment:
```suggestion
    @@multi_tenant_models.push([table_name, model_klass])
```
What I meant to ask is whether there's a case where `table_name` parameter doesn't match `modal_klass.table_name`, in that case you'd store `table_name` in the array so that in `self.multi_tenant_model_for_table` the stored table_name would be used instead of `model.table_name`

I don't know if that scenario would run into issues elsewhere, & even if it doesn't maybe that use should be deprecated.. I'm wanting to merge improvements, release a backwards compatible `1.2.0`, & consider `2.0` as a next release which could include improvements like #125